src/CMakeLists.txt

```diff
@@ -19,3 +19,21 @@ if (CMAKE_CXX_COMPILER)
       target_link_libraries (mongoc-cxx-check PRIVATE mongoc_shared)
    endif ()
 endif ()
+
+# public-header-warnings
+add_executable(public-header-warnings EXCLUDE_FROM_ALL ${CMAKE_CURRENT_SOURCE_DIR}/c-check.c)
+
+target_link_libraries(public-header-warnings PRIVATE mongoc_static)
+
+target_compile_options(public-header-warnings PRIVATE
+  $<$<C_COMPILER_ID:AppleClang,Clang>:-Weverything>
+  $<$<C_COMPILER_ID:GNU>:-Wextra>
+  $<$<C_COMPILER_ID:AppleClang,Clang,GNU>:-Werror>
+  $<$<C_COMPILER_ID:AppleClang,Clang,GNU>:-Wno-declaration-after-statement>
+  $<$<C_COMPILER_ID:AppleClang,Clang,GNU>:-Wno-disabled-macro-expansion>
+  $<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-reserved-identifier>
+  $<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-documentation-unknown-command>
+  $<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-unknown-warning-option>
```

Comment:
Some versions of Clang don't have certain warnings yet, this flag will silence the errors about unknown warnings.